src/relations.rs

```diff
@@ -70,6 +92,128 @@ where
+    /// #     fn set_id(&mut self, id: String) { unimplemented!() }
+    /// # }
+    /// // Build the relation
+    /// let mut brands = CollectionWithId::default();
+    /// let biky_idx = brands.push(Brand {
+    ///     id: "biky".to_string(),
+    /// }).unwrap();
+    /// let mut bikes = CollectionWithId::default();
+    /// let loulou_idx = bikes.push(Bike {
+    ///     id: "loulou".to_string(),
+    ///     brand_id: "biky".to_string(),
+    /// }).unwrap();
+    /// let mut relation = OneToMany::new(&brands, &bikes, "brands_to_bikes").unwrap();
+    ///
+    /// // Add a new bike to the relation
+    /// let fifi_idx = bikes.push(Bike {
+    ///     id: "fifi".to_string(),
+    ///     brand_id: "biky".to_string(),
+    /// }).unwrap();
+    /// relation.add_link(biky_idx, fifi_idx);
+    ///
+    /// // Assert the new relation has been updated
+    /// assert_eq!(
+    ///     relation.get_corresponding_forward(&idx_set![biky_idx]),
+    ///     idx_set![loulou_idx, fifi_idx],
+    /// );
+    /// assert_eq!(
+    ///     relation.get_corresponding_backward(&idx_set![fifi_idx]),
+    ///     idx_set![biky_idx],
+    /// );
+    ///
+    /// // Add a new brand/bike to the relation
+    /// let biclou_idx = brands.push(Brand {
+    ///     id: "biclou".to_string()
+    /// }).unwrap();
+    /// let riri_idx = bikes.push(Bike {
+    ///     id: "riri".to_string(),
+    ///     brand_id: "biclou".to_string(),
+    /// }).unwrap();
+    /// relation.add_link(biclou_idx, riri_idx);
+    ///
+    /// // Assert the new relation has been updated
+    /// assert_eq!(
+    ///     relation.get_corresponding_forward(&idx_set![biclou_idx]),
+    ///     idx_set![riri_idx],
+    /// );
+    /// assert_eq!(
+    ///     relation.get_corresponding_backward(&idx_set![riri_idx]),
+    ///     idx_set![biclou_idx],
+    /// );
+    ///
+    /// // Change an existing relation (fifi was a biky)
+    /// relation.add_link(biclou_idx, fifi_idx);
+    ///
+    /// // Assert the new relations
+    /// assert_eq!(
+    ///     relation.get_corresponding_forward(&idx_set![biclou_idx]),
+    ///     idx_set![fifi_idx, riri_idx],
+    /// );
+    /// assert_eq!(
+    ///     relation.get_corresponding_backward(&idx_set![fifi_idx]),
+    ///     idx_set![biclou_idx],
+    /// );
+    /// // Assert other relations are also updated (fifi is not in biky anymore)
+    /// assert_eq!(
+    ///     relation.get_corresponding_forward(&idx_set![biky_idx]),
+    ///     idx_set![loulou_idx],
+    /// );
+    /// assert_eq!(
+    ///     relation.get_corresponding_backward(&idx_set![loulou_idx]),
+    ///     idx_set![biky_idx],
+    /// );
+    /// ```
+    ///
+    /// # Important
+    ///
+    /// The caller need to ensure that the object referenced by `Idx<U>`
+    /// has a real relation with the object referenced by `Idx<T>`.
+    /// If not, then `Relation` might return Undefined Behavior.
+    pub fn add_link(&mut self, from: Idx<T>, to: Idx<U>) {
```

Comment:
```suggestion
    pub fn add_link(&mut self, from: &T, to: &U) {
```

So you should be able to do `<U as Id<T>>::id(to)` to check whatever needs to be checked.

Reply:
Can you provide a little definition of what is a link (in the documentation).